python/vyos/ifconfig/interface.py

```diff
@@ -339,7 +367,12 @@ def __init__(self, ifname, **kargs):
         self.vrrp = VRRP(ifname)
 
     def _create(self):
-        cmd = 'ip link add dev {ifname} type {type}'.format(**self.config)
+        # Do not create interface that already exist or exists in netns
+        #if self.exists(f'{self.ifname}') or get_interface_namespace(self.config['ifname']):
+        if self.exists(f'{self.ifname}') or _interface_exists_in_netns(self.config['ifname'], self.netns):
```

Comment:
Why not extend `vyos.ifconfig.interface.exists()` and make it netns aware with an optional argument? So it would check if the interface exists either in the global context or the netns context - for any match it will return True

Reply:
Done!
```
    def exists(cls, ifname, netns=None) -> bool:
        cmd = f'ip link show dev {ifname}'
        if netns:
           cmd = f'ip netns exec {netns} {cmd}'
        return run(cmd) == 0

```